package PS;
import "proto/range.proto";

message MatrixInfo {
  enum Type {
    DENSE = 1;
    SPARSE = 2;
    SPARSE_BINARY = 3;
    // TODO gpus
  }
  required Type type = 1;
  required bool row_major = 2;
  // e.g. feature group id
  optional int32 id = 3;

  // size
  required PbRange row = 5;
  required PbRange col = 6;
  // number of non-zero entries
  optional uint64 nnz = 7;

  optional uint32 sizeof_index = 8;
  required uint32 sizeof_value = 9;

  // statisitic
  optional float nnz_per_row = 11;
  optional float nnz_per_col = 12;

  // localization
  // optional bool localized = 21 [default = false];
  // optional uint32 local_rows = 22;
  // optional uint32 local_cols = 23;

  // // stored files
  // repeated string proto_files = 21;
  // repeated string text_files = 22;
  // // a single binary file
  // optional string bin_file = 23;
}
